contracts/crosschain/axelar/AxelarGatewayBase.sol

```diff
@@ -41,8 +43,19 @@ abstract contract AxelarGatewayBase is Ownable {
     error RemoteGatewayAlreadyRegistered(bytes2 chainType, bytes chainReference);
 
     /// @dev Sets the local gateway address (i.e. Axelar's official gateway for the current chain).
-    constructor(IAxelarGateway _gateway) {
+    constructor(IAxelarGateway _gateway, IAxelarGasService _gasService) {
         _axelarGateway = _gateway;
+        _axelarGasService = _gasService;
+    }
+
+    // This is already exposed by AxelarExecutable which AxelarDestinationGateway inherit from. Because its not
+    // virtual, resolution is not possible. Therefore, we should not expose it.
+    // function gateway() public view virtual returns (IAxelarGateway) {
+    //     return _axelarGateway;
+    // }
+
+    function gasService() public view virtual returns (IAxelarGasService) {
```

Comment:
Is this function required by Axelar?

Reply:
No, I thought of that as "discovery". Just like `gateway()` is not required.

Reply:
should we not have it ?